pudl/extract/epacems.py

```diff
@@ -0,0 +1,144 @@
+"""
+Retrieve data from EPA CEMS hourly zipped CSVs.
+
+This modules pulls data from EPA's published CSV files.
+"""
+import glob
+import os
+import pandas as pd
+from pudl import settings
+import pudl.constants as pc
+
+def get_epacems_dir(year):
+    """
+    Data directory search for EPA CEMS hourly
+
+    Args:
+        year (int): The year that we're trying to read data for.
+    Returns:
+        path to appropriate EPA CEMS data directory.
+    """
+    # These are the only years we've got...
+    assert year in range(min(pc.data_years['epacems']),
+                         max(pc.data_years['epacems']) + 1)
+
+    return os.path.join(settings.EPACEMS_DATA_DIR, 'epacems{}'.format(year))
+
+
+def get_epacems_file(year, month, state):
+    """
+    Given a year, month, and state, return the appropriate EPA CEMS zipfile.
+
+    Args:
+        year (int): The year that we're trying to read data for.
+        month (int): The month we're trying to read data for.
+        state (str): The state we're trying to read data for.
+    Returns:
+        path to EPA CEMS zipfiles for that year, month, and state.
+    """
+    state = state.lower()
+    month = str(month).zfill(2)
+    filename = f'epacems{year}{state}{month}.zip'
+    full_path = os.path.join(get_epacems_dir(year), filename)
+    assert os.path.isfile(full_path), (
+        f"ERROR: Failed to find EPA CEMS file for {state}, {year}-{month}.\n" +
+        f"Expected it here: {full_path}")
+    return full_path
+
+
+def fix_names(df):
```

Comment:
For the other data sources, these re-namings have been encoded in `constants.py` I think, though that file is a bit overwhelming.